mobile/src/components/AllBooksWithFilters.tsx

```diff
@@ -0,0 +1,206 @@
+import React, { useState } from 'react';
+import { Text, View, StyleSheet, Dimensions, Pressable, Image } from 'react-native';
+
+import { LoadingCircle } from './LoadingCircle';
+import { PaginationBookList } from './PaginationBookList';
+import { useBookSearchFilter } from './BookSearchFilter';
+import { Book } from '../models/Book';
+import { TextStyles } from '../styles/TextStyles';
+import { Languages } from '../models/Languages';
+import { Colors } from '../styles/Colors';
+
+// how many books to display per page (in the pagination booklist)
+const booksPerPage = 9;
+
+const { width } = Dimensions.get('window');
+
+type AllBooksWithFiltersProps = { allBooks: Book[], loading: boolean};
+
+// ISO 639-1 2-letter language codes (from the Languages file) and the number of languages
+const langCodes = Object.keys(Languages);
+const numLangs = langCodes.length;
+
+// array of objects containing the language code and whether the language is "active" (if its button is pressed)
+// for use in the filter with default of "not active"
+const langArray = [];
+langCodes.forEach(el => {
+  langArray.push({ lang: el, isActive: false });
+});
+
+// object containing the array of language objects for use as a state
+const langState = {
+  arr: langArray,
+};
+
+/**
+ *  Renders the All Books section with a search filter and displays the paginated books.
+ */
+export const AllBooksWithFilters: React.FC<AllBooksWithFiltersProps> = ({ allBooks, loading }) => {
```

Comment:
Yeah I think we can remove this component entirely. All of the dropdown and filtering logic can go into the hook, and the "No results" text can just go into the home screen

Reply:
For the separate component, if I use the hook directly in the HomeScreen, every time a state changes, the screen "jumps" to the top, but moving it to the AllBooksWithFilters makes it so that the state change doesn't move the position of the screen. Is there a way of maintaining the screen position or should I just keep the component?

Reply:
Huh, that shouldn't happen. Maybe there's a moment during the change in state where the list of books is empty, and the view collapses to height=0, shifting the screen? You can get around that by just giving the view a minimum height when there are no books. Just my guess, if that doesn't help try removing this component anyway and I can help debug.